### ISSUE TITLE
README/Documentation Updates

### ISSUE THREAD

**Issue by cartermak** [NASA-AMMOS/aerie-cli#39]
- [ ] Remove any examples that are just copies of help text from the CLI
- [ ] Reconsider how we document specific commands (consider removing the full documentation of the plans/models commands)
- [ ] Consider restructuring some examples into a "Cookbook"
- [ ] Document CLI completion (need `compinit` in .zshrc)

**Comment by cartermak** [comment]
Updated thoughts: 

- We should auto-generate CLI documentation. To start, we can do this with `typer-cli`.
- We should consider auto-generating full documentation of the API using something like `Sphinx`. Libraries that seem to have been added to `aerie-cli` a long time ago but not fully integrated include the `furo` theme and `sphinx-click`.
- Coverage reports can be re-added (with automation, assuming we want them to be used) by importing `coverage` as a dev dependency and configuring in `pyproject.toml`:

```toml
[tool.coverage.paths]
source = ["src", "*/site-packages"]
tests = ["tests", "*/tests"]

[tool.coverage.run]
branch = true
source = ["aerie_cli", "tests"]

[tool.coverage.report]
show_missing = true
fail_under = 100
```

**Comment by cartermak** [comment]
Partially migrated to #76 and will otherwise be closed by #77.
